src/utils/getPreparedPeople.ts

```diff
@@ -0,0 +1,67 @@
+import { Person } from '../types';
+import { SortParams } from '../types/Enums';
+
+function checkCenturyOfLive(centuries: string[], person: Person) {
+  return centuries.some(century => {
+    const centuryEnd = +(century + '00');
+    const centuryStart = centuryEnd - 100;
+
+    if (centuryStart <= +person.born && +person.born < centuryEnd) {
+      return true;
+    }
+
+    return false;
+  });
+}
+
+export const getPreparedPeople = (
+  people: Person[],
+  sex: string | null,
+  query: string | null,
+  centuries: string[],
+  activeSort: string | null,
+  isDesc: string | null,
+) => {
+  let preparedPeople = [...people];
+  const preparedQuery = query?.trim().toLowerCase();
+
+  if (sex) {
+    preparedPeople = preparedPeople.filter(person => person.sex === sex);
+  }
+
+  if (preparedQuery) {
+    preparedPeople = preparedPeople.filter(
+      person =>
+        person.name.toLowerCase().includes(preparedQuery) ||
+        person.fatherName?.toLowerCase().includes(preparedQuery) ||
+        person.motherName?.toLowerCase().includes(preparedQuery),
+    );
+  }
```

Comment:
u can use `some` method for it, something like:
```suggestion
  if (preparedQuery) {
   preparedPeople = preparedPeople.filter(person => {
    const { name, fatherName = '', motherName = '' } = person;
    
    return [name, fatherName, motherName].some(field =>
      field.toLowerCase().includes(preparedQuery)
    );
   });
  }
```
It will be more readable